numpy/core/tests/examples/cython/checks.pyx

```diff
@@ -77,3 +77,88 @@ def make_iso_8601_datetime(dt: "datetime"):
         cnp.NPY_NO_CASTING,
     )
     return result
+
+
+cdef cnp.NpyIter* npyiter_from_npyiter_obj(object it):
+    # this function supports up to 3 iterator operands
+    cdef:
+        cnp.NpyIter* cit
+        # cnp.PyArray_Descr* op_dtypes[5]
+        cnp.npy_uint32 op_flags[3]
+        cnp.PyArrayObject* ops[3]
+        cnp.npy_uint32 flags = 0
+
+    if it.has_index:
+        flags |= cnp.NPY_ITER_C_INDEX
+    if it.has_delayed_bufalloc:
+        flags |= cnp.NPY_ITER_BUFFERED | cnp.NPY_ITER_DELAY_BUFALLOC
+    if it.has_multi_index:
+        flags |= cnp.NPY_ITER_MULTI_INDEX
+
+    # one of READWRITE, READONLY and WRTIEONLY at the minimum must be specified for op_flags
+    for i in range(it.nop):
+        op_flags[i] = cnp.NPY_ITER_READONLY
+
+    for i in range(it.nop):
+        # op_dtypes[i] = cnp.PyArray_DESCR(it.operands[i])
+        ops[i] = <cnp.PyArrayObject*>it.operands[i]
+
+    cit = cnp.NpyIter_MultiNew(it.nop, &ops[0], flags, cnp.NPY_KEEPORDER,
+                               cnp.NPY_NO_CASTING, &op_flags[0],
+                               <cnp.PyArray_Descr**>NULL)
+    return cit
+
+
+def get_npyiter_size(it: "nditer"):
+    cdef cnp.NpyIter* cit = npyiter_from_npyiter_obj(it)
```

Comment:
maybe `from_nditer_obj`?  But OK either way.

Reply:
Yes, that is what I meant to write, thanks for pointing that out.